homeassistant/components/renault/binary_sensor.py

```diff
@@ -0,0 +1,86 @@
+"""Support for Renault sensors."""
+from __future__ import annotations
+
+from renault_api.kamereon.enums import ChargeState, PlugState
+
+from homeassistant.components.binary_sensor import (
+    DEVICE_CLASS_BATTERY_CHARGING,
+    DEVICE_CLASS_PLUG,
+    BinarySensorEntity,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .renault_entities import RenaultBatteryDataEntity, RenaultDataEntity
+from .renault_hub import RenaultHub
+from .renault_vehicle import RenaultVehicleProxy
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Renault entities from config entry."""
+    proxy: RenaultHub = hass.data[DOMAIN][config_entry.entry_id]
+    entities = get_entities(proxy)
+    async_add_entities(entities)
+
+
+def get_entities(proxy: RenaultHub) -> list[RenaultDataEntity]:
+    """Create Renault entities for all vehicles."""
+    entities = []
+    for vehicle in proxy.vehicles.values():
+        entities.extend(get_vehicle_entities(vehicle))
+    return entities
+
+
+def get_vehicle_entities(vehicle: RenaultVehicleProxy) -> list[RenaultDataEntity]:
+    """Create Renault entities for single vehicle."""
+    entities: list[RenaultDataEntity] = []
+    if "battery" in vehicle.coordinators:
+        entities.append(RenaultPluggedInSensor(vehicle, "Plugged In"))
+        entities.append(RenaultChargingSensor(vehicle, "Charging"))
+    return entities
```

Comment:
```suggestion
async def async_setup_entry(
    hass: HomeAssistant,
    config_entry: ConfigEntry,
    async_add_entities: AddEntitiesCallback,
) -> None:
    """Set up the Renault entities from config entry."""
    proxy: RenaultHub = hass.data[DOMAIN][config_entry.entry_id]

    entities = []

    for vehicle in proxy.vehicles.values():
        if "battery" in vehicle.coordinators:
            entities.append(RenaultPluggedInSensor(vehicle, "Plugged In"))
            entities.append(RenaultChargingSensor(vehicle, "Charging"))

    async_add_entities(entities)
```

This should be simplified